front/lib/tracking/browser.ts

```diff
@@ -0,0 +1,72 @@
+import type { LightWorkspaceType } from "@dust-tt/types";
+
+import { AmplitudeClientSideTracking } from "@app/lib/tracking/amplitude/browser";
+
+export class ClientSideTracking {
+  static trackPageView({
+    user,
+    pathname,
+    workspaceId,
+  }: {
+    user?: { id: number };
+    pathname: string;
+    workspaceId?: string;
+  }) {
+    return AmplitudeClientSideTracking.trackPageView({
```

Comment:
why return here? these function should return nothing right + we'll have multiple calls eventually?

Reply:
Some of the (already existing) browser  implementation use the promise in order to not leave the page before the tracking is actually flushed (otherwise we miss some events)